config/config-api/src/com/thoughtworks/go/domain/PipelineGroups.java

```diff
@@ -158,18 +158,17 @@ public void addError(String fieldName, String message) {
         configErrors.add(fieldName, message);
     }
 
-    public HashSet<MaterialConfig> getAllUniquePostCommitSchedulableMaterials() {
-        HashSet<MaterialConfig> materialConfigs = new HashSet<MaterialConfig>();
-        Set<Map> uniqueMaterials = new HashSet<Map>();
+    public Set<MaterialConfig> getAllUniquePostCommitSchedulableMaterials() {
+        Set<MaterialConfig> materialConfigs = new HashSet<MaterialConfig>();
+        Set<String> uniqueMaterials = new HashSet<String>();
         for (PipelineConfigs pipelineConfigs : this) {
             for (PipelineConfig pipelineConfig : pipelineConfigs) {
                 for (MaterialConfig materialConfig : pipelineConfig.materialConfigs()) {
-                    if (!uniqueMaterials.contains(materialConfig.getSqlCriteria()) && materialConfig instanceof ScmMaterialConfig) {
-                        if (materialConfig.isAutoUpdate()) {
-                            continue;
-                        }
+                    if ((materialConfig instanceof ScmMaterialConfig || materialConfig instanceof PluggableSCMMaterialConfig)
+                            && !materialConfig.isAutoUpdate()
+                            && !uniqueMaterials.contains(materialConfig.getFingerprint())) {
                         materialConfigs.add(materialConfig);
-                        uniqueMaterials.add(materialConfig.getSqlCriteria());
+                        uniqueMaterials.add(materialConfig.getFingerprint());
```

Comment:
1. `PluggableSCMMaterialConfig#getFingerprint()` is not locally cached, as with other materials.

Same as `PackageDefinition#getFingerprint()`. They are cached to handle the case of plugin not available. The fingerprint will change based on wether plugin is available or not (in the later case all fields are treated as part of identity).

2. PluggableSCMMaterialConfig#getFingerprint() returning null

will check if that is possible & how it can be handled.